win32/src/win32evtlog.i

```diff
@@ -1010,11 +1212,116 @@ static PyObject *PyEvtUpdateBookmark(PyObject *self, PyObject *args, PyObject *k
 
+PyObject *PyList_FromEVT_VARIANTArray( PEVT_VARIANT val )
+{
+	if ( (val->Type & EVT_VARIANT_TYPE_ARRAY) == 0){
+		PyErr_SetString(PyExc_RuntimeError, "Trying to create a list from an EVT_VARIANT that is not an array");
+		return NULL;
+	}
+
+	PyObject* ret = PyList_New( val->Count );
+	DWORD val_type = val->Type & EVT_VARIANT_TYPE_MASK;
+	for ( DWORD i = 0; i < val->Count; ++i ) {
+		PyObject *obval = NULL;
+		switch (val_type){
+			case EvtVarTypeString:
+				obval = PyWinObject_FromWCHAR(val->StringArr[i]);
+				break;
+			case EvtVarTypeAnsiString:
+				obval = PyWinCoreString_FromString(val->AnsiStringArr[i]);
+				break;
+			case EvtVarTypeSByte:
+				obval = PyLong_FromLong(val->SByteArr[i]);
+				break;
+			case EvtVarTypeByte:
+				obval = PyLong_FromUnsignedLong(val->ByteArr[i]);
+				break;
+			case EvtVarTypeInt16:
+				obval = PyLong_FromLong(val->Int16Arr[i]);
+				break;
+			case EvtVarTypeUInt16:
+				obval = PyLong_FromUnsignedLong(val->UInt16Arr[i]);
+				break;
+			case EvtVarTypeInt32:
+				obval = PyLong_FromLong(val->Int32Arr[i]);
+				break;
+			case EvtVarTypeUInt32:
+				obval = PyLong_FromUnsignedLong(val->UInt32Arr[i]);
+				break;
+			case EvtVarTypeInt64:
+				obval = PyLong_FromLongLong(val->Int64Val);
+				break;
+			case EvtVarTypeUInt64:
+				obval = PyLong_FromUnsignedLongLong(val->UInt64Arr[i]);
+				break;
+			case EvtVarTypeSingle:
+				obval = PyFloat_FromDouble(val->SingleArr[i]);
+				break;
+			case EvtVarTypeDouble:
+				obval = PyFloat_FromDouble(val->DoubleArr[i]);
+				break;
+			case EvtVarTypeBoolean:
+				obval = PyBool_FromLong(val->BooleanArr[i]);
+				break;
+			case EvtVarTypeGuid:
+				obval = PyWinObject_FromIID(val->GuidArr[i]);
+				break;
+			case EvtVarTypeSizeT:
+				obval = PyInt_FromSsize_t(val->SizeTArr[i]);
+				break;
+			case EvtVarTypeFileTime:
+				{
+				// FileTimeVal is defined as ULONGLONG but
+				// FileTimeArr is defined as an array of FILETIME
+				LARGE_INTEGER timestamp;
+				timestamp.LowPart = val->FileTimeArr[i].dwLowDateTime;
+				timestamp.HighPart = val->FileTimeArr[i].dwHighDateTime;
+				obval = PyWinObject_FromTimeStamp(timestamp);
+				break;
+				}
+			case EvtVarTypeSysTime:
+				obval = PyWinObject_FromSYSTEMTIME(val->SysTimeArr[i]);
+				break;
+			case EvtVarTypeSid:
+				obval = PyWinObject_FromSID(val->SidArr[i]);
+				break;
+			case EvtVarTypeHexInt32:
+				{
+				PyObject* number = PyLong_FromUnsignedLong( val->UInt32Arr[i] );
+				obval = PyNumber_ToBase( number, 16 );
+				Py_XDECREF( number );
+				break;
+				}
+			case EvtVarTypeHexInt64:
+				{
+				PyObject* number = PyLong_FromUnsignedLongLong( val->UInt64Arr[i] );
+				obval = PyNumber_ToBase( number, 16 );
+				Py_XDECREF( number );
+				break;
+				}
+			case EvtVarTypeEvtXml:
+				obval = PyWinObject_FromWCHAR(val->XmlValArr[i]);
+				break;
+			default:
+				PyErr_Format(PyExc_NotImplementedError, "EVT_VARIANT_TYPE %d not supported yet", val_type);
+				break;
+		    }
+
```

Comment:
Same comment as the loop above - and in this case you are formatting an error but not using it.